pagai/views.py

```diff
@@ -93,6 +89,32 @@ def state(database_name):
         return jsonify({"status": "unknown or training"})
 
 
+@api.route("/explore/<credential_id>/<table>", methods=["GET"])
+def explore(credential_id, table):
+    """
+    Database exploration: returns the first rows of
+    a database table. The db credentials are retrieved from
+    Pyrog. The number of returned rows may be specified using
+    query params (eg: /explore/<db>/<table>?first=10).
+    """
+    credentials = pyrog.get_credentials(credential_id)
+    limit = request.args.get("first", 10, type=int)
+    schema = request.args.get("schema")
+
+    def explore_postgres():
+        connection = postgres.from_pyrog_credentials(credentials)
```

Comment:
nit: do you think we could easily be able to write
```
with postgres.from_pyrog_credentials(credentials) as connection:
    res = postgres.explore(table, connection=connection, limit=limit, schema=schema)
```
which is a bit more Pythonic and where we don't need to explicitly close the connection?

Reply:
oh I forgot about the fact that you could omit to explicitly close the connection this way, indeed!